x-pack/plugin/core/src/test/java/org/elasticsearch/license/AbstractLicensesIntegrationTestCase.java

```diff
@@ -88,7 +88,7 @@ public void onFailure(String source, @Nullable Exception e) {
     protected void assertLicenseActive(boolean active) throws Exception {
         assertBusy(() -> {
             for (XPackLicenseState licenseState : internalCluster().getDataNodeInstances(XPackLicenseState.class)) {
-                if (licenseState.isActive() == active) {
+                if (licenseState.allowForAllLicenses() == active) {
```

Comment:
I don't think this call should change. I think we still need a (possibly package protected) `isActive` method here so that the test is asserting what it says it's asserting.

Reply:
Added package private [isActive](https://github.com/elastic/elasticsearch/pull/52118/files#diff-75ffc27a1f769f6d57256b3a4f04babcR355). But having both of the two methods bothers me a bit. They are functionally identical. But it could be accidental. Given we should fallback to basic license which is always active, the call to `allowForAllLicenses` is really needed. It can either be removed or reduced to always return true.